modules/apps/redirect/redirect-test/src/testIntegration/java/com/liferay/redirect/internal/upgrade/v3_0_2/test/UpgradeRedirectEntrySourceURLTest.java

```diff
@@ -0,0 +1,276 @@
+		RedirectEntry redirectEntry4 =
+			RedirectEntryLocalServiceUtil.getRedirectEntry(redirectEntryId4);
+		RedirectEntry redirectEntry5 =
+			RedirectEntryLocalServiceUtil.getRedirectEntry(redirectEntryId5);
+		RedirectEntry redirectEntry6 =
+			RedirectEntryLocalServiceUtil.getRedirectEntry(redirectEntryId6);
+
+		Assert.assertEquals("page1", redirectEntry1.getSourceURL());
+		Assert.assertEquals("page2", redirectEntry2.getSourceURL());
+		Assert.assertEquals("page3", redirectEntry3.getSourceURL());
+		Assert.assertEquals("page1", redirectEntry4.getSourceURL());
+		Assert.assertEquals("page2", redirectEntry5.getSourceURL());
+		Assert.assertEquals("page3", redirectEntry6.getSourceURL());
+	}
+
+	@Test
+	public void testUpgradeWithSingleGroup() throws Exception {
+		long redirectEntryId1 = _counterLocalService.increment();
+		long redirectEntryId2 = _counterLocalService.increment();
+		long redirectEntryId3 = _counterLocalService.increment();
+
+		addRedirectEntry(
+			redirectEntryId1, _group1.getGroupId(), _group1.getCompanyId(),
+			"destination1", "PageA");
+		addRedirectEntry(
+			redirectEntryId2, _group1.getGroupId(), _group1.getCompanyId(),
+			"destination2", "PageB");
+		addRedirectEntry(
+			redirectEntryId3, _group1.getGroupId(), _group1.getCompanyId(),
+			"destination3", "PageC");
+
+		_redirectEntrySourceURLUpgradeProcess.upgrade();
+
+		RedirectEntry redirectEntry1 =
+			RedirectEntryLocalServiceUtil.getRedirectEntry(redirectEntryId1);
+		RedirectEntry redirectEntry2 =
+			RedirectEntryLocalServiceUtil.getRedirectEntry(redirectEntryId2);
+		RedirectEntry redirectEntry3 =
+			RedirectEntryLocalServiceUtil.getRedirectEntry(redirectEntryId3);
+
+		Assert.assertEquals("pagea", redirectEntry1.getSourceURL());
+		Assert.assertEquals("pageb", redirectEntry2.getSourceURL());
+		Assert.assertEquals("pagec", redirectEntry3.getSourceURL());
+	}
+
+	@Test
+	public void testUpgradeWithSourceURLCollision() throws Exception {
+		if (_db.getDBType() == DBType.MYSQL) {
+			Assume.assumeTrue(
+				"MYSQL is case-insensitive so it is not possible to create " +
+					"this scenario, skip test.",
+				_db.getDBType() == DBType.MYSQL);
+		}
+		else {
+			long redirectEntryId1 = _counterLocalService.increment();
+			long redirectEntryId2 = _counterLocalService.increment();
+
+			addRedirectEntry(
+				redirectEntryId1, _group1.getGroupId(), _group1.getCompanyId(),
+				"destination1", "testpage");
+			addRedirectEntry(
+				redirectEntryId2, _group1.getGroupId(), _group1.getCompanyId(),
+				"destination2", "TestPage");
+
+			_redirectEntrySourceURLUpgradeProcess.upgrade();
+
+			RedirectEntry redirectEntry1 =
+				RedirectEntryLocalServiceUtil.getRedirectEntry(
+					redirectEntryId1);
+			RedirectEntry redirectEntry2 =
+				RedirectEntryLocalServiceUtil.getRedirectEntry(
+					redirectEntryId2);
+
+			Assert.assertEquals("testpage", redirectEntry1.getSourceURL());
+			Assert.assertEquals("TestPage", redirectEntry2.getSourceURL());
+		}
```

Comment:
Hi @adolfopa, 
When I run the test locally on MySQL db the code was executed and threw an error as it is not possible to insert the second row because MySQL is case insensitive. So I put it in a try-catch to avoid executing the second half.

Reply:
@bakayattila The whole point of `Assume` is to skip a test if the condition is false. If you put `Assume.assumeFalse("...", _db.getDBType() == DBType.MYSQL)` at the beginning of your test it will be skipped if you're on MySQL. The conditional is not necessary.

Reply:
I changed to code according to your suggestion.